src/Metadata/Property/Factory/SerializerPropertyMetadataFactory.php

```diff
@@ -108,41 +112,40 @@ private function transformLinkStatus(ApiProperty $propertyMetadata, array $norma
             return $propertyMetadata;
         }
 
-        // TODO: 3.0 support multiple types, default value of types will be [] instead of null
-        $type = $propertyMetadata->getBuiltinTypes()[0] ?? null;
+        $types = $propertyMetadata->getBuiltinTypes() ?? [];
 
-        if (null === $type) {
-            return $propertyMetadata;
-        }
+        foreach ($types as $type) {
+            if (
+                $type->isCollection() &&
+                $collectionValueType = $type->getCollectionValueTypes()[0] ?? null
+            ) {
+                $relatedClass = $collectionValueType->getClassName();
+            } else {
+                $relatedClass = $type->getClassName();
+            }
 
-        if (
-            $type->isCollection() &&
-            $collectionValueType = $type->getCollectionValueTypes()[0] ?? null
-        ) {
-            $relatedClass = $collectionValueType->getClassName();
-        } else {
-            $relatedClass = $type->getClassName();
-        }
+            // if property is not a resource relation, don't set link status (as it would have no meaning)
+            if (null === $relatedClass || !$this->isResourceClass($relatedClass)) {
+                return $propertyMetadata;
```

Comment:
what if another `type` is a resource class? We should first compute every types before making a decision.